scrape/scrape.go

```diff
@@ -1134,7 +1137,9 @@ loop:
 			}
 
 			var ref uint64
-			ref, err = app.Add(lset, t, v)
+			if err != errSampleLimit {
```

Comment:
Shouldn't we be checking that errSampleLimit is not nil?

A comment here would also be useful.

Reply:
I've put a second test that will be more effective and only do something if we breach the limit.